soc/arm/renesas_smartbond/da1469x/Kconfig.defconfig.series

```diff
@@ -17,4 +17,15 @@ config SYS_CLOCK_HW_CYCLES_PER_SEC
 config SRAM_VECTOR_TABLE
 	default y
 
+config USE_DT_CODE_PARTITION
+	default y if MCUBOOT
+
+config FLASH_BASE_ADDRESS
+	default 0
+	default 0x16000000 if MCUBOOT
+	default 0x16000000 if BOOTLOADER_MCUBOOT
```

Comment:
For this SoC it's just more convenient to link at 0x0. Depending on product header, the offset of application image on flash may vary. If application is linked at 0x0 can simply take .bin and write it regardless of product header contents as internal bootloader will remap proper flash offset to 0x0. If using 0x16000000 as a base address we need to know the load offset in advance. This is the case for mcuboot and the reason why boot partition starts at 0x2400 since this is expected load offset with default product header written by ezFlashCLI.

So basically it should be possible to always set `FLASH_BASE_ADDRESS=0x16000000` (or set from dts) but `USE_DT_CODE_PARTITION=n` would also require `FLASH_LOAD_OFFSET=0x2400` to write after "default" product header.

Reply:
other arm based SoC generally uses the value from devicetree, so just for that I believe it would be nice to stick to fetching default value based on devicetree as it will make arm based SoCs more consistent with each other.

And because you will be taking the FLASH_LOAD_OFFSET based on devicetree, but the FLASH_BASE_ADDRESS directly from Kconfig. This gives a slight mismatch wrt. where and how flash and code partition values are defined and obtained.
Mixing flash defaults specied in devicetree with defaults specified directly in Kconfig is always a recipe for confusion.
Ref: https://github.com/zephyrproject-rtos/zephyr/blob/7119cc160531fdd0439c5404cb5d991e44efd98f/Kconfig.zephyr#L100-L103

But afaik there is no official guide that dictate how flash base and flash load offset must be obtained / that it must be obtained  in a specific manner, so I leave the final decision to the maintainer.

Reply:
I updated `FLASH_BASE_ADDRESS` to be set from dts. `FLASH_LOAD_OFFSET` has default value overridden only if `USE_DT_CODE_PARTITION=n` due to mentioned earlier requirements of internal bootloader.